packages/terra-list/src/ListSectionHeader.scss

```diff
@@ -0,0 +1,93 @@
+    border-left: var(--terra-list-section-header-border-left);
+    border-right: var(--terra-list-section-header-border-right);
+    border-top: var(--terra-list-section-header-border-top);
+    box-shadow: var(--terra-list-section-header-box-shadow);
+    color: var(--terra-list-section-header-color, #1c1f21);
+    padding-bottom: var(--terra-list-section-header-padding-bottom, 0.5rem);
+    padding-top: var(--terra-list-section-header-padding-top, 0.5rem);
+    width: 100%;
+
+    &.is-interactable {
+      @include terra-padding-start(var(--terra-list-section-header-accordion-padding-start, 0.5rem));
+      cursor: pointer;
+
+      &:hover {
+        background: var(--terra-list-section-header-hover-background, #d7d8d9);
+        border-bottom: var(--terra-list-section-header-hover-border-bottom, 1px solid #c8cacb);
+        border-left: var(--terra-list-section-header-hover-border-left);
+        border-right: var(--terra-list-section-header-hover-border-right);
+        border-top: var(--terra-list-section-header-hover-border-top);
+        box-shadow: var(--terra-list-section-header-hover-box-shadow);
+        color: var(--terra-list-section-header-hover-color, #1c1f21);
+      }
+
+      &:focus {
+        background: var(--terra-list-section-header-focus-background, #e8e9ea);
+        border-bottom: var(--terra-list-section-header-focus-border-bottom, 1px solid #c8cacb);
+        border-left: var(--terra-list-section-header-focus-border-left);
+        border-right: var(--terra-list-section-header-focus-border-right);
+        border-top: var(--terra-list-section-header-focus-border-top);
+        box-shadow: var(--terra-list-section-header-focus-box-shadow, 0 0 1px 3px rgba(76, 178, 233, 0.5), 0 0 7px 4px rgba(76, 178, 233, 0.35));
+        color: var(--terra-list-section-header-focus-color, #1c1f21);
+        outline: none;
+      }
+
+      &:active,
+      &.is-active {
+        background: var(--terra-list-section-header-active-background, #d7d8d9);
+        border-bottom: var(--terra-list-section-header-active-border-bottom, 1px solid #c8cacb);
+        border-left: var(--terra-list-section-header-active-border-left);
+        border-right: var(--terra-list-section-header-active-border-right);
+        border-top: var(--terra-list-section-header-active-border-top);
+        box-shadow: var(--terra-list-section-header-active-box-shadow);
+        color: var(--terra-list-section-header-active-color, #1c1f21);
+      }
+    }
+  }
+
+  .title {
+    display: inline;
+    font-size: var(--terra-list-section-header-font-size, 1rem);
+    font-weight: var(--terra-list-section-header-font-weight, bold);
+    line-height: var(--terra-list-section-header-line-height, 1.142857143);
+    text-transform: var(--terra-list-section-header-text-transform);
+  }
+
+  .accordion-icon-wrapper {
+    @include terra-margin-start(var(--terra-list-section-header-accordion-icon-margin-start));
```

Comment:
This should work without the `terra-margin-start` mixin. Try setting this `margin-left: var(--terra-list-section-header-accordion-icon-margin-start);` and testing in RTL. We use a postcss RTL plugin that should handle transforming it. 
This can be done for the margin-end, padding-start, and padding-end mixins.